packages/salesforcedx-vscode-replay-debugger/src/states/variableAssignmentState.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { ApexVariable } from '../adapter/apexReplayDebug';
+import { LogContext } from '../core/logContext';
+import { DebugLogState } from './debugLogState';
+
+export class VariableAssignmentState implements DebugLogState {
+  private fields: string[];
+
+  constructor(fields: string[]) {
+    this.fields = fields;
+  }
+  public handle(logContext: LogContext): boolean {
+    const currFrame = logContext.getTopFrame();
+    if (currFrame) {
+      const id = currFrame.id;
+      const frameInfo = logContext.getFrameHandler().get(id);
+      const name = this.fields[3];
+      const nameSplit = this.fields[3].split('.');
+      const varName =
```

Comment:
On a begin scope for local variables, only the variable name is given. On an assignment, the classname and variable name are given. We stored the variable name as the key in the map so we need to retrieve just that piece to update the local variable.